File: v3-sandbox/prototype-java/src/main/java/org/hiero/keys/io/KeyFormat.java
package org.hiero.keys.io;

import java.util.Objects;
import org.hiero.keys.KeyType;
import org.jspecify.annotations.NonNull;

/**
 * Combined container format and encoding. Mirrors the meta-language enum
 * {@code keys.io.KeyFormat}.
 */
public enum KeyFormat {

    PKCS8_WITH_DER(KeyContainer.PKCS8, KeyEncoding.DER),
    SPKI_WITH_DER(KeyContainer.SPKI, KeyEncoding.DER),
    PKCS8_WITH_PEM(KeyContainer.PKCS8, KeyEncoding.PEM),
    SPKI_WITH_PEM(KeyContainer.SPKI, KeyEncoding.PEM);

    private final KeyContainer container;

    private final KeyEncoding encoding;

    KeyFormat(@NonNull final KeyContainer container, @NonNull final KeyEncoding encoding) {
        this.container = container;
        this.encoding = encoding;
    }

    /**
     * @return the container of this format
     */
    @NonNull
    public KeyContainer getContainer() {
        return container;
    }

    /**
     * @return the encoding of this format
     */
    @NonNull
    public KeyEncoding getEncoding() {
        return encoding;
    }

    /**
     * @return the raw format produced by this format's encoding
     */
    @NonNull
    public RawFormat getRawFormat() {
        return encoding.getRawFormat();
    }

    /**
     * Returns whether this format can hold a key of the given type.
     *
     * @param type the key type to check
     * @return {@code true} if the underlying container supports the given key type
     */
    public boolean supportsType(@NonNull final KeyType type) {
        Objects.requireNonNull(type, "type must not be null");
        return container.supportsType(type);
    }

    /**
     * Decodes the given string value into raw bytes for the given key type.
     *
     * @param keyType the key type the value belongs to
     * @param value   the encoded string value
     * @return the decoded raw bytes
     */
    @NonNull
    public byte[] decode(@NonNull final KeyType keyType, @NonNull final String value) {
        Objects.requireNonNull(keyType, "keyType must not be null");
        Objects.requireNonNull(value, "value must not be null");
        return encoding.decode(keyType, value);
    }
}

File: v3-sandbox/prototype-python/src/hiero_sdk_v3/keys/io/key_format.py
from enum import Enum
from .key_container import KeyContainer
from .key_encoding import KeyEncoding
from .raw_format import RawFormat
from ..key_type import KeyType


class KeyFormat(Enum):
    def __init__(self, container: KeyContainer, encoding: KeyEncoding) -> None:
        self.container = container
        self.encoding = encoding

    PKCS8_WITH_DER = (KeyContainer.PKCS8, KeyEncoding.DER)
    SPKI_WITH_DER = (KeyContainer.SPKI, KeyEncoding.DER)
    PKCS8_WITH_PEM = (KeyContainer.PKCS8, KeyEncoding.PEM)
    SPKI_WITH_PEM = (KeyContainer.SPKI, KeyEncoding.PEM)

    @property
    def raw_format(self) -> RawFormat:
        return self.encoding.raw_format

    def supports_type(self, key_type: KeyType) -> bool:
        return self.container.supports_type(key_type)
